### DISCUSSION TITLE
Magic Link + Magic Phrase = 🧡

### DISCUSSION THREAD

**Discussion by barrelltech** [supabase/supabase#9132] [Feature Requests]
I was using magic link for people to log in, but it's very difficult to support in it PWA / Home Screen Web Apps. It would be awesome if a temporary passphrase/word/number was available with the email so users could return to the PWA and enter in the code, instead of copying the insanely long confirmationURL

**Comment by GaryAustin1** [comment]
It is.  I use it with my home screen app.
You use {{token}} in the email to provide the token.
```
Please enter this code in the confirmation box in the app.

<h2>{{.Token}}</h2>
```
I have them signUp, which sends the confirmation token to their email.

Then go to a screen with an input for the code.

When they enter the code I do:
```
            const {data, error} = await supabase.auth.verifyOtp({
                email: email,
                token: confirmValue.trim(),
                type: "signup"
            })
```

Documented here for v2.
https://supabase.com/docs/reference/javascript/next/auth-verifyotp

It worked on v1 with same call I believe.

**Reply by barrelltech** [reply]
If I try to use `{{ .Token }}` in my email, it fails to save. Do I need to update supabase somehow, or tell it to use the v2 library?

**Reply by barrelltech** [reply]
Oddly, if I use the local instance of supabase, the default email is delivered with a code

<img width="452" alt="Screen Shot 2022-09-24 at 16 14 04" src="https://user-images.githubusercontent.com/99540338/192102738-9b28a5f8-c00d-425e-a454-908a859c2346.png">

**Reply by barrelltech** [reply]
~Fwiw I was able to randomly guess the correct variable of `{{ .Code }}`~

While this saved, it did not work. I tried `{{ .Token }}` again and it worked. I guess I was just getting unfortunately timed unrelated errors ¯\_(ツ)_/¯
